examples/data_preprocessing/[WIP]_chexpert_dataset/Prepare_CheXpert_data.ipynb

```diff
@@ -0,0 +1,958 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class CheXpertDatasetProcessor(Dataset):\n",
+    "    \n",
+    "    def __init__(self, \n",
+    "                 path: str,\n",
+    "                 subset: str, \n",
+    "                 image_paths: List[str], \n",
+    "                 number_of_images: int,  \n",
+    "                 transform_sequence: list = None,\n",
+    "                 to_ones: List[str] = None,\n",
+    "                 to_zeros: List[str] = None, \n",
+    "                 to_ignore: List[str] = None,\n",
+    "                 return_image: bool = True):\n",
+    "        \n",
+    "        \"\"\"\n",
+    "        Args:\n",
+    "            path: path to the folder where train.csv and valid.csv are stored\n",
+    "            subset: either \"train\" to load the train.csv or \"valid\" to load valid.csv\n",
+    "            image_paths: paths to the images\n",
+    "            number_of_images: number of images in the dataset\n",
+    "            transform_sequence: sequence used to transform the images\n",
+    "            to_ones: list of pathologies for which uncertainty labels should be replaced by 1\n",
+    "            to_zeros: list of pathologies for which uncertainty labels should be replaced by 0\n",
+    "            to_ignore: list of pathologies for which uncertainty labels should be ignored (label will be turned to nan)\n",
+    "            return_image: True: image tensor and labels are returned, False: only labels are returned\n",
+    "        Returns: \n",
+    "            224 x 224 image tensor and a corresponding tensor containing 12 labels\n",
+    "        \"\"\"\n",
+    "        \n",
+    "        self.path = path\n",
+    "        self.subset = subset\n",
+    "        self.image_paths = image_paths\n",
+    "        self.number_of_images = number_of_images\n",
+    "        self.transform_sequence = transform_sequence\n",
+    "        self.to_ones = to_ones\n",
+    "        self.to_zeros = to_zeros\n",
+    "        self.to_ignore = to_ignore\n",
+    "        self.return_image = return_image\n",
+    "        \n",
+    "    def process_chexpert_dataset(self):\n",
+    "        \n",
+    "        # read dataset\n",
+    "        if self.subset == \"train\":\n",
+    "            data = pd.read_csv(\"train.csv\")\n",
+    "            \n",
+    "        elif self.subset == \"valid\":\n",
+    "            data = pd.read_csv(\"valid.csv\")\n",
+    "            \n",
+    "        else:\n",
+    "            raise ValueError(\"Invalid subset, please choose either 'train' or 'valid'\")\n",
+    "            \n",
+    "        pathologies = data.iloc[:, -13:-1].columns\n",
+    "        \n",
+    "        # prepare labels\n",
+    "        data.iloc[:, -13:-1] = data.iloc[:, -13:-1].replace(float(\"nan\"), -1) # blank labels -> uncertain\n",
+    "        \n",
+    "        if self.to_ones:\n",
```

Comment:
Oh I'm sorry, I have corrected it wrongly the last time. Ofc here and further it should be `if self.to_ones is not None` as you had had it before. I didn't get the logic of these parameters and thought they were booleans.